### PR TITLE
Add unit test for func BoolEqual

### PR DESCRIPTION
**Pull request by Hellcatlk**
Signed-off-by: zouyu <zouy.fnst@cn.fujitsu.com>

### CODE REVIEW

File: pkg/internal/assert/assert_test.go [hunk 1]
```diff
@@ -64,6 +64,34 @@ func TestExpectError(t *testing.T) {
...
+func TestBoolEqual(t *testing.T) {
+	t.Parallel()
+	t.Run("not equal", func(t *testing.T) {
+		t.Parallel()
+		var f fakeT
+		BoolEqual(&f, true, false)
+		if int(f) == 0 {
+			t.Fatalf("Expected t.Errorf to be called but it was not")
+		}
+		BoolEqual(&f, false, true)
```
Comment (neolit123) [inline pkg/internal/assert/assert_test.go:76]: second test case with the swapped arguments can be omitted?

File: pkg/internal/assert/assert_test.go [hunk 2]
```diff
@@ -64,6 +64,34 @@ func TestExpectError(t *testing.T) {
...
+		}
+	})
+	t.Run("equal", func(t *testing.T) {
+		t.Parallel()
+		var f fakeT
+		BoolEqual(&f, true, true)
+		if int(f) != 0 {
+			t.Fatalf("Expected t.Errorf not to be called but it was")
+		}
+		BoolEqual(&f, false, false)
```
Comment (neolit123) [inline pkg/internal/assert/assert_test.go:88]: same question here.

File: pkg/internal/assert/assert_test.go (hunk 1 above)
Comment (Hellcatlk) [reply pkg/internal/assert/assert_test.go]: Thanks, done.

File: pkg/internal/assert/assert_test.go (hunk 2 above)
Comment (Hellcatlk) [reply pkg/internal/assert/assert_test.go]: Thanks, done.

**Comment by BenTheElder** [comment]
e2e tests aren't necessary here. saving CI resources.

**Pull request merged**